src/gluonts/transform.py

```diff
@@ -155,6 +155,42 @@ def __call__(self, ts: np.ndarray, a: int, b: int) -> np.ndarray:
         return np.array([b])
 
 
+class ContinuousTimePointSampler:
+    """
+    Abstract class for "continuous time" samplers, which, given a lower bound
+    and upper bound, sample "points" (events) in continuous time from a
+    specified interval.
+    """
+
+    def __init__(self, num_instances: int) -> None:
+        self.num_instances = num_instances
+
+    def __call__(self, a: float, b: float) -> np.ndarray:
+        """
+        Returns random points in the real interval between :code:`a` and
+        :code:`b`.
+
+        Parameters
+        ----------
+        a
+            The lower bound (minimum value that the sampled point can take)
+        b
+            Upper bound. Must be greater than a.
+        """
+        raise NotImplementedError()
+
+
+class ContinuousTimeUniformSampler(ContinuousTimePointSampler):
+    """
+    Implements a simple random sampler to sample points in the continuous
+    interval between :code:`a` and :code:`b`.
+    """
+
+    def __call__(self, a: float, b: float) -> np.ndarray:
+        assert a <= b
```

Comment:
add an error message here?

Reply:
That and a few other validations in the module are now more verbose.